Sources/MapboxMaps/Gestures/GestureHandlers/PinchGestureHandler.swift

```diff
@@ -30,16 +33,22 @@ internal class PinchGestureHandler: GestureHandler {
             scale = pow(2, delegate.scaleForZoom())
             delegate.gestureBegan(for: .pinch)
 
+            previousPinchCenterPoint = pinchCenterPoint
+
             /**
              TODO: Handle a concurrent rotate gesture here.
              Prioritize the correct gesture by comparing the velocity of competing gestures.
              */
-
         } else if pinchGestureRecognizer.state == .changed {
 
             let newScale = scale * pinchGestureRecognizer.scale
-            delegate.pinchScaleChanged(with: log2(newScale), andAnchor: pinchCenterPoint)
+            let newZoom = log2(newScale)
 
+            let offset = CGSize(width: pinchCenterPoint.x - previousPinchCenterPoint.x,
+                                height: pinchCenterPoint.y - previousPinchCenterPoint.y)
+            previousPinchCenterPoint = pinchCenterPoint
```

Comment:
This class uses a pinch gesture, but I think the principle here is the same as the [recommendation](https://developer.apple.com/documentation/uikit/uipangesturerecognizer/1621207-translation) when using a pan gesture:

> The x and y values report the total translation over time. They are not delta values from the last time that the translation was reported. Apply the translation value to the state of the view when the gesture is first recognized—do not concatenate the value each time the handler is called.

Would it be possible to refactor this so that we always calculate the new camera options relative to the camera options when the gesture began, not relative to the camera options at the previous frame?

Reply:
There is a desire to move this implementation to use the dragAPI, instead of manually calculating a new camera center position. That API has a method to retrieve cameraOptions for the drag operation and it needs the center from the previous frame along with the new center point. That said, I think we may end up wanting to stick with something similar to what we have here. I will circle back to this though as things become clearer. I'll also be adding some additional notes below to describe the challenges related to the DragAPI.

`func dragCameraOptions(from: CGPoint, to: CGPoint) -> CameraOptions`

Reply:
The approach changed and I'm not sure the original comment still applies. The DragAPI requires the to/from screen points in order to handle change from the previous frame and drag location. We're no longer needing to calculate offsets either.